src/Voron/Data/PostingLists/NativeIntegersList.cs

```diff
@@ -145,13 +145,23 @@ public void SortAndRemoveDuplicatesAndRemovals()
         Count = (int)(outputBufferPtr - RawItems + 1);
     }
 
-    public int MoveTo(Span<long> matches)
+    public int MoveTo(Span<long> output)
     {
-        var read = Math.Min(Count, matches.Length);
-        new Span<long>(this.RawItems, read).CopyTo(matches);
-        Count -= read;
-        RawItems += read;
-        return read;
+        var count = Math.Min(Count, output.Length);
+        new Span<long>(RawItems, count).CopyTo(output);
+
+        // Check if we have moved the entire content.
+        if (Count == count)
+        {
+            Count = 0;
+            return count;
+        }
+
+        var theRestToCopy = new Span<long>(RawItems + count, Count - count);
+        theRestToCopy.CopyTo(new Span<long>(RawItems, Capacity));
+        Count = theRestToCopy.Length;
+
```

Comment:
I don't really like this - the issue is that we are now copying twice, and this is something that is used in high perf location.
Maybe better to have a *window* into this that would not impact the actual content?

Reply:
Could be an option. I am rewriting all this anyways with the cleanup of lists. but this is the correct behavior that has to be implemented at the level of the list itself. Those changes have to be done at the level of the caller.